git_machete/client.py

```diff
@@ -1656,6 +1657,56 @@ def flush_caches(self) -> None:
         self.__branch_defs_by_sha_in_reflog = None
         self.__git.flush_caches()
 
+    def checkout_github_pr(self, pr_no: int) -> None:
+        org: str
+        repo: str
+        remote: str
+        remote, (org, repo) = self.__derive_remote_and_github_org_and_repo()
+        debug('checkout_github_pr', f'organization is {org}, repository is {repo}')
+
+        all_prs: List[GitHubPullRequest] = derive_pull_requests(org, repo)
+        pr: GitHubPullRequest = check_pr_already_created_by_number(pr_no, all_prs)
+
+        if not pr:
+            raise MacheteException(f"PR number {pr_no} is not found in repository {repo}")
+        debug('checkout_github_pr', f'found {pr}')
+
+        self.__git.fetch_remote(remote)
+
+        path: List[str] = self.get_path(pr.head)
+        reversed_path: List[str] = path[::-1]  # need to add from root do down
+        for index, branch in enumerate(reversed_path):
+            if branch not in self.managed_branches:
+                if index == 0:
+                    self.__cli_opts.opt_as_root = True
+                    self.add(branch)
+                    self.__cli_opts.opt_as_root = False
+                    continue
+                self.__cli_opts.opt_onto = reversed_path[index - 1]
+                self.add(branch)
+
+        self.sync_annotations_to_github_prs()
+
+        self.__git.checkout(pr.head)
+        print(fmt(f"Pull request #{pr.number} checked out at local branch {pr.head}"))
+
+    def get_path(self, last_branch: str) -> List[str]:
+        def get_parent_branch_from_pr(branch: str) -> Optional[str]:
+            for pr in all_prs:
+                if pr.head == branch:
+                    return pr.base
+            return None
+
+        path: List[str] = []
+        current_branch: str = last_branch
+        remote, (org, repo) = self.__derive_remote_and_github_org_and_repo()
+        all_prs: List[GitHubPullRequest] = derive_pull_requests(org, repo)
```

Comment:
Hmmm where's `all_prs` used?

Reply:
in nested function `get_parent_branch_from_pr` at line 1695

Reply:
Whoops okay, that's pretty brittle/surprising (that the only reference to a local var is **before** its declaration)... I think the `get_parent_branch_from_pr` method can be replaced with a one-liner using `next(filter(pred, iterable), None)`